### PR TITLE
Site selector needs to update on login/out as

### PR DESCRIPTION
**Pull request by alfredo-dotcms**

### CODE REVIEW

File: src/app/view/components/login-as/login-as.ts [hunk 1]
```diff
@@ -70,6 +72,7 @@ export class LoginAsComponent extends BaseComponent implements OnInit {
                     this.close();
                     this.iframeOverlayService.hide();
                     this.dotNavigationService.goToFirstPortlet();
+                    this.dotEventsService.notify('login-as');
```
Comment (freddyucv) [inline src/app/view/components/login-as/login-as.ts]: Why if we have 'his.loginService. loginAs()' we need re tigger the event here?

File: src/app/view/components/toolbar-user/toolbar-user.ts [hunk 2]
```diff
@@ -64,6 +66,7 @@ export class ToolbarUserComponent extends BaseComponent implements OnInit {
                 this.dropdown.closeIt();
                 this.iframeOverlayService.hide();
                 this.dotNavigationService.goToFirstPortlet();
+                this.dotEventsService.notify('logout-as');
```
Comment (freddyucv) [inline src/app/view/components/toolbar-user/toolbar-user.ts]: Why if we have 'his.loginService.logoutAs()' we need re tigger the event here?

File: src/app/view/components/_common/site-selector/site-selector.component.spec.ts [hunk 3]
```diff
@@ -13,36 +13,49 @@ import { SearchableDropdownComponent } from '../searchable-dropdown/component/se
...
+        'should send notification when login-as/logout-as',
+        fakeAsync(() => {
+            const dotEventsService = de.injector.get(DotEventsService);
+            spyOn(comp, 'getSitesList');
+            fixture.detectChanges();
+            dotEventsService.notify('login-as');
+            tick(0);
+            dotEventsService.notify('logout-ass');
+            tick(0);
+            expect(comp.getSitesList).toHaveBeenCalledTimes(2);
```
Comment (freddyucv) [inline src/app/view/components/_common/site-selector/site-selector.component.spec.ts]: this should be two test

File: src/app/view/components/_common/site-selector/site-selector.component.ts [hunk 4]
```diff
@@ -67,6 +68,11 @@ export class SiteSelectorComponent implements OnInit, OnChanges, OnDestroy {
         this.refreshSitesSub = this.siteService.refreshSites$.subscribe((_site: Site) => this.handleSitesRefresh());
 
         this.getSitesList();
+        ['login-as', 'logout-as'].map((event: string) => {
```
Comment (fmontes) [inline src/app/view/components/_common/site-selector/site-selector.component.ts]: This should be a foreach because map return a new array.

**Pull request merged**